### PR TITLE
Refactor(UI): Refactor lineage component code

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on the Refactor lineage component code 

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx [hunk 1]
```diff
@@ -1355,14 +1049,49 @@ const EntityLineageComponent: FunctionComponent<EntityLineageProp> = ({
...
+      {showDeleteModal && (
+        <Modal
+          okText={
+            deletionState.loading ? (
+              <Loader size="small" type="white" />
+            ) : deletionState.status === 'success' ? (
+              <FontAwesomeIcon className="tw-text-white" icon="check" />
+            ) : (
+              <span>Confirm</span>
+            )
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx:1084]: Can we avoid nested ternary conditions?

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.interface.ts [hunk 2]
```diff
@@ -83,8 +83,17 @@ export interface SelectedEdge {
 
 export type ElementLoadingState = Exclude<LoadingState, 'waiting'>;
 
-export type CustomeElement = { node: Node[]; edge: FlowEdge[] };
-export type CustomeFlow = Node | FlowEdge;
+export type CustomElement = { node: Node[]; edge: FlowEdge[] };
+export type CustomFlow = Node | FlowEdge;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.interface.ts:87]: Thanks, @ShaileshParmar11 , for taking care of this.

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 3]
```diff
@@ -515,3 +528,181 @@ export const getNodeRemoveButton = (onClick: () => void) => {
...
+  edgeData: EdgeData
+) => {
+  return edgeArr.filter(
+    (edge) =>
+      !edgeArr.find(
+        () =>
+          edgeData.fromId === edge.fromEntity && edgeData.toId === edge.toEntity
+      )
+  );
+};
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:543]: Can we add comments for each util method?

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 4]
```diff
@@ -515,3 +528,181 @@ export const getNodeRemoveButton = (onClick: () => void) => {
...
+    (!isNil(sourceDownstreamNode) && !isNil(targetUpStreamNode)) ||
+    !isNil(sourceDownstreamNode) ||
+    !isNil(targetDownStreamNode) ||
+    source?.includes(entity.id);
+
+  if (isUpstream) {
+    return 'upstream';
+  } else if (isDownstream) {
+    return 'downstream';
+  }
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:684]: We have two types of nodes `upstream` and `downstream` so can we create an enum for that and use it in all the places?

File: openmetadata-ui/src/main/resources/ui/src/styles/app.less [hunk 5]
```diff
@@ -132,6 +143,11 @@
   text-align: right;
 }
 
+// position
+.absolute {
+  position: absolute;
+}
+
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/styles/app.less:150]: Already we have this in `position.less`

File: openmetadata-ui/src/main/resources/ui/src/styles/spacing.less [hunk 6]
```diff
@@ -359,3 +363,19 @@
 .w-80 {
   width: 80%;
 }
+// positioning
+
+//top
+.top-1 {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/styles/spacing.less:369]: We have `position.less` can we use that

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 7]
```diff
@@ -515,3 +529,292 @@ export const getNodeRemoveButton = (onClick: () => void) => {
...
+      )
+  );
+};
+
+/**
+ * Finds the upstream/downstream edge based on selected edge
+ * @param edgeArr edge[]
+ * @param data selected edge
+ * @returns edge
+ */
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:551]: Can we add unit tests for new functions?

**Comment by ShaileshParmar11** [comment]
Addressed all the comments and added unit test @chirag-madlani @Sachin-chaurasiya

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 8]
```diff
@@ -515,3 +529,292 @@ export const getNodeRemoveButton = (onClick: () => void) => {
...
+        sourceHandle: sourceHandle,
+        targetHandle: targetHandle,
+      },
+    };
+  }
+
+  return data;
+};
+
+export const LoadingStatus = (
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:808]: The function name should be in `camelCase`.
```suggestion
export const getLoadingStatus = (
```

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**